applications/ConvectionDiffusionApplication/tests/bfecc_convection_test.py

```diff
@@ -100,34 +112,47 @@ def runTest(self):
     def checkResults(self):
         with UnitTest.WorkFolderScope(self.work_folder, __file__):
             if self.print_reference_values:
-                with open(self.reference_file + '.csv','w') as reference_file:
-                    reference_file.write("#ID, TEMPERATURE\n")
-                    for node in self.main_model_part.Nodes:
-                        reference_file.write("{0}, {1}\n".format(node.Id, node.GetSolutionStepValue(KratosMultiphysics.TEMPERATURE, 0)))
-                    reference_file.close()
+                json_output_settings = KratosMultiphysics.Parameters(r'''{
+                        "output_variables": ["TEMPERATURE"],
+                        "output_file_name": "",
+                        "model_part_name": "MainModelPart",
+                        "time_frequency": 0.0
+                }''')
+                json_output_settings["output_file_name"].SetString(self.reference_file + "_results.json")
```

Comment:
this will not run when called from the core test-runner, you need the `GetFilePath` that we usually use
=> remember that this is needed for all reference files

Reply:
this should be solved for good in #5577